apps/channel/components/share/view.coffee

```diff
@@ -0,0 +1,113 @@
+Backbone = require 'backbone'
+copy = require 'copy-to-clipboard'
+{ extend, map } = require 'underscore'
+{ APP_URL, CURRENT_PATH } = require('sharify').data
+template = -> require('./index.jade') arguments...
+
+module.exports = class ChannelShareView extends Backbone.View
+  className: 'ChannelShare'
+
+  events:
+    'click .js-toggle': 'toggle'
+    'click .js-copy-url': 'copyURL'
+    'click .js-share-external': 'shareExternal'
+    'click .js-reload': 'reload'
+    'click .js-disable-public-link': 'disablePublicLink'
+    'click .js-select-all': 'selectAll'
+
+  initialize: ({ channel, text }) ->
+    @channel = channel
+    @state = new Backbone.Model
+      status: 'inactive'
+      url: APP_URL + CURRENT_PATH
+      text: encodeURIComponent(text)
+
+    @listenTo @state, 'change', @render
+
+  toggle: (e) ->
+    e.preventDefault()
+
+    if @channel.get('status') is 'private' and not @channel.has('share_link')
+      @state.set 'status', 'generating'
+
+      @channel
+        .generateShareLink()
+        .then () =>
+          @state.set
+            status: 'active'
+            url: @channel.shareHref()
+
+        .catch () =>
+          @state.set 'status', 'error'
+
+    else if @channel.get('status') is 'private'
```

Comment:
Am I following this right? If a channel already has a share link, it just shows the menu

Reply:
Yeah it just skips generation.